tests/source/test_interpolate.f90

```diff
@@ -0,0 +1,103 @@
+program test_interpolate
+    use math_constants
+    use util
+
+    implicit none
+
+    real(8), parameter :: TOL = 1.0d-6
+
+    real(8), parameter :: X_MIN = 0.0d0, X_MAX = TWOPI
+
+    call test_spline_1d(spline_order=3, periodic=.False.)
+    call test_spline_1d(spline_order=3, periodic=.True.)
+    call test_spline_1d(spline_order=4, periodic=.False.)
+    call test_spline_1d(spline_order=4, periodic=.True.)
+    call test_spline_1d(spline_order=5, periodic=.False.)
+    call test_spline_1d(spline_order=5, periodic=.True.)
+
+
+    call test_spline_2d(spline_order=(/3,3/), periodic=(/.False., .False./))
```

Comment:
The array constructor syntax `[3, 3]` is [available since Fortran 2003 and supported by GNU Fortran 4.4 since 2008-05-16](https://gcc.gnu.org/wiki/Fortran2003Status).